app/src/main/java/com/example/dogbook/main/fragments/MapFragment.java

```diff
@@ -43,12 +66,70 @@ private void setUpMap() {
             mapFragment.getMapAsync(new OnMapReadyCallback() {
                 @Override
                 public void onMapReady(GoogleMap googleMap) {
-                    loadMap(googleMap);
+                    loadedMap(googleMap);
                 }
             });
         }
     }
 
-    private void loadMap(GoogleMap googleMap) {
+    private void loadedMap(GoogleMap googleMap) {
+        map = googleMap;
+        //Once the map is loaded, set up the current location
+        getPermissionsToCurrentLocation();
+    }
+
+    @NeedsPermission({Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION})
+    private void getPermissionsToCurrentLocation() {
+        //Checks permissions
+        if (ContextCompat.checkSelfPermission(getContext(), Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissions(new String[] {Manifest.permission.ACCESS_FINE_LOCATION}, 1);
+            return;
+        }
+        getCurrentLocation();
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (requestCode == 1) {
+            if (grantResults[0] != PackageManager.PERMISSION_DENIED) {
+                //all permissions have been granted
+                getCurrentLocation();
+                return;
+            }
+            //If permissions are not granted, current location will not be available
+            Toast.makeText(getContext(), "Permissions to get current location denied", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    //Permission already granted in getPermissionsToCurrentLocation()
+    @SuppressLint("MissingPermission")
+    private void getCurrentLocation() {
+        locationManager = (LocationManager) getActivity().getSystemService(LOCATION_SERVICE);
+        locationListener = new LocationListener() {
+            @Override
+            public void onLocationChanged(@NonNull Location location) {
+                currentLocation = location;
+                displayLocation(); //Map camera focuses on changed location
```

Comment:
Hm, do you want this to change each time the user's location changes or just set it the first time?